Continue on the development of features

Xhz0729 opened the pull request:
This pull request includes several enhancements and new features to the client application, focusing on integrating Google Maps, adding a booking widget, and improving the place form functionality. The most important changes include the addition of new dependencies, the introduction of new components, and updates to existing components to support new features.

### New Features and Enhancements:

* **Dependencies:**
  * Added `@react-google-maps/api` to `client/package.json` and `client/package-lock.json` for Google Maps integration. [[1]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR11) [[2]](diffhunk://#diff-1846122c2c83a486a3693f7966aa522c34cf489f674185c4da0d9221683fd81fR13)
  * Added `@googlemaps/js-api-loader`, `@googlemaps/markerclusterer`, and other related dependencies to `client/package-lock.json`. [[1]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR897-R912) [[2]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR1085-R1114) [[3]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR1400-R1405)

* **Components:**
  * Introduced `BookingWidget` component to handle booking functionalities, including displaying price, check-in/out dates, and guest number input.
  * Added `MapComponent` to display a Google Map with a marker for the place's address.

* **App Component:**
  * Updated `client/src/App.jsx` to include new routes for `HomePage` and `PlaceDetailsPage`. [[1]](diffhunk://#diff-51b002bf85fbd146b38871b89ba0d41c2bead37756295748c67206aac23ddc9aR4-R12) [[2]](diffhunk://#diff-51b002bf85fbd146b38871b89ba0d41c2bead37756295748c67206aac23ddc9aR24-R31)

* **AddPlaceForm Component:**
  * Enhanced `AddPlaceForm` with new state for price and functions to delete and select photos as the main photo. [[1]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aR21-R22) [[2]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aR116-R135) [[3]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aL166-R249)
  * Improved the layout and added an input for price per night.

These changes collectively enhance the application's functionality, providing a more comprehensive user experience with new features like map integration and booking management.This pull request introduces several new dependencies and features to the `client` package, including Google Maps integration, a booking widget, and additional functionality for handling photos in the `AddPlaceForm` component.

### New Dependencies:
* Added `@react-google-maps/api` to integrate Google Maps into the application. (`client/package-lock.json`, `client/package.json`) [[1]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR11) [[2]](diffhunk://#diff-1846122c2c83a486a3693f7966aa522c34cf489f674185c4da0d9221683fd81fR13)
* Included `@googlemaps/js-api-loader`, `@googlemaps/markerclusterer`, and other related packages to support Google Maps features. (`client/package-lock.json`) [[1]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR897-R912) [[2]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR1085-R1114) [[3]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR1400-R1405)
* Added `invariant` and `supercluster` for additional utility functions. (`client/package-lock.json`) [[1]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR3290-R3298) [[2]](diffhunk://#diff-925cb809ce77c7f7f7f8bbd7594593438fc006dc56bf432a624c47722a784d3dR5205-R5213)

### New Features:
* **Google Maps Integration:**
  * Created `MapComponent` to display a map based on the place's address. (`client/src/components/MapComponent.jsx`)
* **Booking Widget:**
  * Developed `BookingWidget` component to handle booking details and user interactions. (`client/src/components/BookingWidget.jsx`)

### Enhancements to `AddPlaceForm`:
* Added functionality to delete photos and select a main photo. (`client/src/components/AddPlaceForm.jsx`) [[1]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aR116-R135) [[2]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aL166-R249)
* Introduced a price field to the form. (`client/src/components/AddPlaceForm.jsx`) [[1]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aR21-R22) [[2]](diffhunk://#diff-87673caf0852692332c6331de6e38c2126b80557c4043aeb200ce91e6dc5e16aR346-R355)
* Updated layout and input handling for check-in, check-out times, and guest numbers. (`client/src/components/AddPlaceForm.jsx`)

### Routing Updates:
* Added routes for `HomePage` and `PlaceDetailsPage` in the `App` component. (`client/src/App.jsx`) [[1]](diffhunk://#diff-51b002bf85fbd146b38871b89ba0d41c2bead37756295748c67206aac23ddc9aR4-R12) [[2]](diffhunk://#diff-51b002bf85fbd146b38871b89ba0d41c2bead37756295748c67206aac23ddc9aR24-R31)

Xhz0729 committed:
Add trash icon near each photo

Xhz0729 committed:
Create the deletePhoto function to handle delete photo button clicked event

Xhz0729 committed:
Add choose main photo feature and prevent button click default behavior

Xhz0729 committed:
Render HomePage

Xhz0729 committed:
Add route to fetch all the places

Xhz0729 committed:
Create HomePage component and list all places existing in database

Xhz0729 committed:
Add price to my Place schema

Xhz0729 committed:
Update the route to include price attribute

Xhz0729 committed:
Add the price state and input

Xhz0729 committed:
Render price on page

Xhz0729 committed:
Add react route for PlaceDetailsPage

Xhz0729 committed:
Add endpoint to fetch the place details by place id

Xhz0729 committed:
Add Link to the place details page

Xhz0729 committed:
Create BookingWidget component

Contain price info, check In&Out input && No. of guests input

Xhz0729 committed:
Install react-google-maps/api package

Xhz0729 committed:
Create my MapComponent

Xhz0729 committed:
Add endpoint for calling Google Map API to geocoding my address

Xhz0729 committed:
Install axios package

ericadev commented on client/src/components/MapComponent.jsx:
```diff
@@ -0,0 +1,80 @@
+        const response = await axios.get(
+          `/geocode?address=${encodeURIComponent(placeData.address)}`
+        );
+        setCoordinates(response.data); // { lat: 29.9974362, lng: -98.0986204 }
```
Remove unnecessary comment

Xhz0729 replied:
> Remove unnecessary comment

Unnecessary comment being removed.

Xhz0729 committed:
Implement PlaceDetailsPage component

- Fetch place details using place ID from URL - Display main image with clickable thumbnail previews for more photos - Include place description, check-in/out times, and maximum guest capacity - Add BookingWidget for reservation functionality - Integrate MapComponent to show the location on Google Maps - Handle loading state while fetching coordinates for the map

Xhz0729 committed:
Get rid of unnecessary comment
